service/storage_service.cc

```diff
@@ -830,6 +830,84 @@ void storage_service::scan_cdc_generations() {
+            if (!_gossiper.is_normal(addr))  {
+                throw std::runtime_error(format("All nodes must be in NORMAL state while performing check_and_repair_cdc_streams"
+                        " ({} is in state {})", addr, _gossiper.get_gossip_status(state)));
+            }
+
+            const auto ts = cdc::get_streams_timestamp_for(addr, _gossiper);
+            if (!latest || (ts && *ts > *latest)) {
+                latest = ts;
+            }
+        }
+
+        bool should_regenerate = false;
+        std::optional<cdc::topology_description> gen;
+        try {
+            gen = _sys_dist_ks.local().read_cdc_topology_description(
+                    *latest, { _token_metadata.count_normal_token_owners() }).get0();
+        } catch (exceptions::request_timeout_exception& e) {
+            throw; // Stop and inform the API caller about timeout
+        } catch (exceptions::unavailable_exception& e) {
+            throw; // Stop and inform the API caller about unavailability
+        } catch (...) {
+            const auto ep = std::current_exception();
+            if (is_timeout_exception(ep)) {
+                throw; // Stop and inform the API caller about other timeout errors
+            }
+            // On exotic errors proceed with regeneration
+            cdc_log.error("Exception while reading CDC topology description: \"{}\". Regenerating streams anyway.", ep);
+            should_regenerate = true;
+        }
+
+        if (!gen) {
+            cdc_log.error(
+                "Could not find CDC generation with timestamp {} in distributed system tables (current time: {}),"
+                " even though some node gossiped about it.",
+                latest, db_clock::now());
+            should_regenerate = true;
+        } else {
+            std::unordered_set<dht::token> gen_ends;
+            for (const auto& entry : gen->entries()) {
+                gen_ends.insert(entry.token_range_end);
+            }
+            for (const auto& metadata_token : _token_metadata.sorted_tokens()) {
+                if (!gen_ends.count(metadata_token)) {
+                    should_regenerate = true;
+                    break;
+                }
+            }
+        }
+
+        if (!should_regenerate) {
+            if (latest != _cdc_streams_ts) {
+                do_handle_cdc_generation(*latest);
+            }
+            cdc_log.info("CDC generation does not need repair");
+            return;
+        }
+        const auto new_streams_ts = cdc::make_new_cdc_generation(db().local().get_config(),
+                {}, _token_metadata, _gossiper,
+                _sys_dist_ks.local(), get_ring_delay(), false /* for_testing */);
+        // Need to artificially update our STATUS so other nodes handle the timestamp change
+        auto status = _gossiper.get_application_state_ptr(get_broadcast_address(), application_state::STATUS);
+        if (!status) {
+            return;
```

Comment:
`!status` should never happen, but we can be defensive and say something in the log (using `slogger` too, because it wouldn't be a CDC specific problem), e.g.:
```
slogger.error("Our STATUS is missing");
cdc_log.error("Aborting CDC generation repair due to missing STATUS");
return;
```

Reply:
done